generator/feature_segments_checker/feature_segments_checker.cpp

```diff
@@ -0,0 +1,220 @@
+#include "generator/srtm_parser.hpp"
+
+#include "routing/bicycle_model.hpp"
+
+#include "coding/file_name_utils.hpp"
+
+#include "geometry/mercator.hpp"
+#include "geometry/point2d.hpp"
+
+#include "indexer/classificator_loader.hpp"
+#include "indexer/feature.hpp"
+#include "indexer/feature_processor.hpp"
+#include "indexer/map_style_reader.hpp"
+
+#include "platform/platform.hpp"
+
+#include "base/logging.hpp"
+
+#include "3party/gflags/src/gflags/gflags.h"
+
+#include "std/iostream.hpp"
+#include "std/map.hpp"
+#include "std/string.hpp"
+
+DEFINE_string(srtm_path, "", "Path to directory with SRTM files");
+DEFINE_string(mwm_path, "", "Path to an mwm file.");
+
+namespace
+{
+routing::BicycleModel const & GetBicycleModel()
+{
+  static routing::BicycleModel const instance;
+  return instance;
+}
+
+template <typename K, typename V>
+void IncrementStats(K const & k, map<K, V> & m)
+{
+  if (m.find(k) == m.cend())
+    m[k] = 0;
+  m[k] += 1;
+}
+
+template <typename Cont>
+void PrintCont(Cont const & cont, string const & title, string const & msgText1,
+               string const & msgText2)
+{
+  cout << endl << title << endl;
+  for (auto const a : cont)
+    cout << a.second << msgText1 << a.first << msgText2 << endl;
+}
+
+/// \brief Adds |point| to |uniqueRoadPoints| if there's no a point in |uniqueRoadPoints|
+/// in |kErrorMeters| environment of |point|.
+/// \note This method does get an exact result. The problem is points are sorted in
```

Comment:
Are you sure that for the set lower_bound and upper_bound will return you different results?
And, in any case, the whole method works quite BADLY. Imagine that the set has following points:
```
(0, 0)
(0000.1, 10000000)
(1000, 1000)
```
And the new point is (0.0002, 0). Could you please guess what lower_bound or upper_bound will return you, and how it's related to the Euclidian metric?